dart.yaml

```diff
@@ -0,0 +1,142 @@
+package:
+  name: dart
+  version: 3.1.1
+  epoch: 0
+  description: "Dart is a client-optimized language for fast apps on any platform"
+  copyright:
+    - license: BSD-3-Clause
+  dependencies:
+    runtime:
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - curl
+      - dart-stage0
+      - gcc
+      - git
+      - gn
+      - llvm-lld-16
+      - llvm16
+      - patch
+      - python-3.11
+      - samurai
+      - teapot-tools
+      - zip
+      - zlib-dev
+      - zstd
+
+pipeline:
+  - runs: |
+      cat <<EOT >> .gclient
+      solutions = [{
+        'name': 'sdk',
+        'url': 'https://dart.googlesource.com/sdk.git@${{package.version}}',
+      }]
+      target_cpu = ['x64', 'arm64']
+      target_cpu_only = True
+      EOT
+
+      gclient sync --no-history --nohooks --tpot-cipd-ignore-platformed
+
+  # apply patches (src is in sdk folder hence we cannot use patch action from melange)
+  - runs: |
+      cp -R *.patch sdk/
+      cd sdk
+      patch '-p1' < no-werror.patch
+      patch '-p1' < gcc-13.patch
+      patch '-p1' < build-config.patch
+
+  - working-directory: sdk
+    runs: |
+      rm -rf tools/sdks/dart-sdk
+      ln -s /usr/lib/dart tools/sdks/dart-sdk
+
+      mkdir -p buildtools/ninja
+      ln -s /usr/bin/gn buildtools/gn
+      ln -s /usr/bin/samu buildtools/ninja/ninja
+
+      python3 tools/generate_package_config.py
+      python3 tools/generate_sdk_version_file.py
+
+      echo '' > tools/bots/dartdoc_footer.html
+      rm third_party/devtools/web/devtools_analytics.js
+
+  - if: ${{build.arch}} == "aarch64"
+    runs: |
```

Comment:
It would be better to fix any hardcoded triplet assumptions in the build instead of doing this.  This will fail when we transition to rootless builds.